src/comm.cc

```diff
@@ -555,13 +556,6 @@ comm_import_opened(const Comm::ConnectionPointer &conn,
 
     comm_init_opened(conn, note, AI);
 
-    if (conn->local.port() > (unsigned short) 0) {
-#if _SQUID_WINDOWS_
-        if (AI->ai_socktype != SOCK_DGRAM)
-#endif
-            fd_table[conn->fd].flags.nolinger = true;
```

Comment:
This particular removal looks like a double bug[^1] to me, but it is actually _correct_: comm_import_opened() is meant to mimic _effects_ of opening a file descriptor in another kid process. As commit a4c0f9c6 (that added this code) explains, both processes point to the same kernel socket object/state associated with process-specific descriptors. This code correctly set (evidently unused) flag to mimic those effects _without_ calling commSetNoLinger(); the other process had already called commSetNoLinger() for the underlying socket.

[^1]: As if we have forgotten to call commSetNoLinger() earlier and are removing all traces of our intent to do so now.